content-moderation-evaluation/Evaluating-Amazon-Rekognition-Content-Moderation-Service.ipynb

```diff
@@ -0,0 +1,542 @@
+    "\n",
+    "To work with Ground Truth, this list needs to be converted to a .json file and uploaded to the S3 BUCKET\n",
+    "\n",
+    "_Note: The ordering of the labels or classes in the template governs the class indices that you will see downstream in the output manifest (this numbering is zero-indexed). In other words, the class that appears second in the template will correspond to class \"1\" in the output._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "CLASS_LIST = [\"<moderation_label_1>\", \"<moderation_label_2\", \"<moderation_label_3\", \"<moderation_label_4\", \"<moderation_label_5\", \"Safe_Content\"]\n",
+    "print(\"Label space is {}\".format(CLASS_LIST))\n",
+    "\n",
+    "json_body = {\"labels\": [{\"label\": label} for label in CLASS_LIST]}\n",
+    "with open(\"class_labels.json\", \"w\") as f:\n",
+    "    json.dump(json_body, f)\n",
+    "\n",
+    "s3.upload_file(\"class_labels.json\", BUCKET, EXP_NAME + \"/class_labels.json\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Create instruction template for Ground Truth Workforce\n",
+    "All of your evaluation dataset will be annotated by human annotators. It is critical to provide clear and concise instructions that help the annotators understand what you want to achieve. When used through the AWS Console, Ground Truth helps you create the instructions using a visual wizard. When using the API, you need to create an HTML template for your instructions. Below, we prepare a very simple but effective template and upload it to your S3 bucket.\n",
+    "\n",
+    "_NOTE:_\n",
+    "- _If you use any images in your template (as we do), they need to be publicly accessible. You can enable public access to files in your S3 bucket through the S3 Console, as described in [3 Documentation](https://aws.amazon.com/premiumsupport/knowledge-center/read-access-objects-s3-bucket/). In this case, we need to have X+1 number of images, X equals to number of moderation labels you like to use plus \"safe content\" label. In our example, it's 6._ \n",
```

Comment:
This sounds controversial to me. I understand we need the annotators to be able to access images, but them being publicly accessible seems to be a security issue for customers.

Can we recommend having bucket policies in place to allow the annotators account to access them?

Also,
```suggestion
    "- _If you use any images in your template they need to be publicly accessible. You can enable public access to files in your S3 bucket through the S3 Console, as described in the [S3 Documentation](https://aws.amazon.com/premiumsupport/knowledge-center/read-access-objects-s3-bucket/). In this case, we need to have X+1 number of images, X equals to number of moderation labels you like to use plus \"safe content\" label. In our example, it's 6._ \n",
```